core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatLabelTransform.java

```diff
@@ -0,0 +1,65 @@
+package com.airbnb.aerosolve.core.transforms;
+
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.util.Util;
+
+import java.util.Map;
+
+import com.typesafe.config.Config;
+
+/**
+ * Convert a Float feature to binary LABEL based on threshold value. This is intended for
+ * binary classification where LABEL is either -1 or 1. MergeStrategy describe how existing
+ * LABEL is treated.
+ */
+public class FloatLabelTransform implements Transform {
+  enum MergeStrategy {
+    IGNORE, // ignore existing label
+    KEEP_POSITIVE, // keep positive label
+    KEEP_NEGATIVE, // keep negative label
+    KEEP_ALL // preserve existing label (only replace if observation is un-labeled
+  }
+
+  private String fieldName1;
+  private String key1;
+  private double threshold;
+  private MergeStrategy mergeStrategy;
+
+  @Override
+  public void configure(Config config, String key) {
+    fieldName1 = config.getString(key + ".field1");
+    key1 = config.getString(key + ".key1");
+    threshold = config.getDouble(key + ".threshold");
+    mergeStrategy = MergeStrategy.valueOf(config.getString(key + ".merge").toUpperCase());
+  }
+
+  @Override
+  public void doTransform(FeatureVector featureVector) {
+    Map<String, Double> labelFeature = Util.getOrCreateFloatFeature("LABEL", featureVector.floatFeatures);
+
+    Double label = labelFeature.get("");
+
+    if (mergeStrategy == MergeStrategy.KEEP_ALL && label != null) return;
+
+    Map<String, Double> floatFeature = featureVector.floatFeatures.get(fieldName1);
+    if (floatFeature != null) {
+      Double featureValue = floatFeature.get(key1);
+      if (featureValue != null) {
+        double newLabel = featureValue >= threshold ? 1 : -1;
+
+        if (
```

Comment:
I agree your profiling is more reasonable in that it excluded JVM starting up time. I think it's your judgment call whether to go for the cleaner code or the 20% performance gain in one transformer. It's not clear how much it will eventually contribute in our production jobs.